### PR TITLE
Add code health use case

### PR DESCRIPTION
**Pull request by jasonyavorsky**
This PR is intended to get content from the 25-page use cases master plan document into a useful "one-pager" form in the handbook so it can be an easier-to-use input to planning and other efforts going on related to use cases.

https://deploy-preview-1782--sourcegraph-handbook.netlify.app/strategy-goals/strategy#use-cases
https://deploy-preview-1782--sourcegraph-handbook.netlify.app/strategy-goals/strategy/use_cases/code_health

It was generated by:

1. Bring in overall structure from example provided by @christinaforney at https://docs.google.com/document/d/1rDhPpC9OOeUl9I98t4h2cYnJApe0KvaJUBLEjUtz60c/edit#heading=h.w01qegla9vmg, exclusive of the roadmap section (the roadmap itself will end up in GitHub issues; in the future we can add a generator to this page that automatically brings in the list of roadmap items that are associated with this use case.)
2. Edit down the content from the code health section of the use cases master doc (https://docs.google.com/document/d/1k7w-sgK3T4B_kLi9wuEIvvMdRp4hPd0YtsUt0UpwvsU/edit#) to map to the structure from step 1.

Note that minor grammar edits were applied, but no major ideas or concepts were changed to avoid making too many different changes at once. The goal here is to get the use case in the handbook, and once that's done repeat the same steps for the rest of the use cases. After that, continuous iterations to improve the content itself can come in using our normal process/tooling for iterating on things in the handbook.

### CODE REVIEW

File: content/strategy-goals/strategy/use_cases/code_health.md [hunk 1]
```diff
@@ -0,0 +1,45 @@
+# Code Health use case
+
+This page provides a strategic introduction to one of our [company use cases](../index.md#use-cases). Check that link to find the rest of the use cases and learn how we use them as part of our company strategy.
+
+## Overall vision
+
+Sourcegraph provides a dashboard source of truth for overall code structure and health. It makes visible the impact of changes on health, and provides interventions for improving code health. We envision this covering rigorous, explicitly measurable aspects of code health such as scorecards and checklists (for example, https://www.cortex.io/products/scorecard and https://www.opslevel.com/checks/ but for code), compliance scoring, and measurable, actionable tech debt. We can also extend this analysis to the dependencies in your software supply chain.
```
Comment (jasonyavorsky) [inline content/strategy-goals/strategy/use_cases/code_health.md]:
```suggestion
Our vision for this use case is that Sourcegraph provides a dashboard source of truth for overall code structure and health. It makes visible the impact of changes on health, and provides interventions for improving code health. We envision this covering rigorous, explicitly measurable aspects of code health such as scorecards and checklists (for example, https://www.cortex.io/products/scorecard and https://www.opslevel.com/checks/ but for code), compliance scoring, and measurable, actionable tech debt. We can also extend this analysis to the dependencies in your software supply chain.
```

**Review by christinaforney — APPROVED** [review]
This looks great! I love that you pulled in customer quotes. I would also love in the near future, to include a demo path example: this is a simple straightforward walk through of the use case primary workflow (this is originally what I had in mind when we talked about adding 'how we solve it today' section).

**Comment by jasonyavorsky** [comment]
> This looks great! I love that you pulled in customer quotes. I would also love in the near future, to include a demo path example: this is a simple straightforward walk through of the use case primary workflow (this is originally what I had in mind when we talked about adding 'how we solve it today' section).

Great feedback @christinaforney, and it made me also realize this would benefit from some instructions so I added this idea as well as some other guidance in HTML comments to the document. Anything else you'd add?

File: content/strategy-goals/strategy/use_cases/code_health.md [hunk 2]
```diff
@@ -0,0 +1,55 @@
+# Code Health use case
+
+This page provides a strategic introduction to one of our [company use cases](../index.md#use-cases). Check that link to find the rest of the use cases and learn how we use them as part of our company strategy.
+
+## Overall vision
+
+<!-- Convey what things will be like in the future for your use case, be as descriptive as you can to help someone understand where we are headed with our vision. -->
```
Comment (jasonyavorsky) [inline content/strategy-goals/strategy/use_cases/code_health.md]:
```suggestion
<!-- Convey what things will be like in the future for your use case, being as descriptive as you can to help someone understand where we are headed with our vision. -->
```

File: content/strategy-goals/strategy/use_cases/code_health.md [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+- Sourcegraph helps us with technical debt reduction and the consolidation of our codebase by letting us avoid duplication, spot the usage of deprecated APIs or internal (non-API) library code, and identify general purpose code in specific projects (such as utility classes) that can be factored out and shared in a core library.
+- With the help of Sourcegraph, we were able to quickly look at all clients of an API and remove unused attributes that lived in different repositories, ultimately simplifying our APIs and speeding up developer iteration time.
+- Our teams are constantly innovating and building new systems, necessitating decomposing and migrating off of older ones. Sourcegraph gives us the ability to search for and refactor references to deprecated services, libraries, URL patterns, and more across our 2000+ repositories, and the confidence that we're not leaving anyone behind.
+
+## Who benefits
+
+<!-- Link to the personas that relate to this use case, and describe how it bnenefits each of them -->
```
Comment (jasonyavorsky) [inline content/strategy-goals/strategy/use_cases/code_health.md]:
```suggestion
<!-- Link to the personas that relate to this use case, and describe how it benefits each of them -->
```

File: content/strategy-goals/strategy/use_cases/code_health.md [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+
+Tech debt is one component in code health that plays a big role in developer experience and productivity, and refers to the cost of refactoring or redesigning code tomorrow because of short-cuts taken today. [Stripe found](https://stripe.com/files/reports/the-developer-coefficient.pdf) that developers spend 33% of their time on tech debt; maintenance of legacy systems and tech debt being the number one cause for productivity loss. For a 50-person engineering team of $100k/year developers, that 33% respresents $1.65M/year. Beyond the monetary implications, organizations bogged down with it often suffer from low morale and churn.
+
+## How we solve this today
+
+<!-- Describe in as much detail as you can how the product enables this use case today. You can include customer quotes, textual walkthroughs, and this is also a great place to link to demo videos. -->
```
Comment (jasonyavorsky) [inline content/strategy-goals/strategy/use_cases/code_health.md]:
```suggestion
<!-- Describe in as much detail as you can how the product enables this use case today. You can include customer quotes, textual walkthroughs, and this is also a great place to link to demo videos. This is perhaps the most important single section in this document, so don't be afraid to add too much - if you feel this section is getting long, consider summarizing here and linking out to other pages in the handbook with details. -->
```

File: content/strategy-goals/strategy/use_cases/code_health.md [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+- Sourcegraph helps us with technical debt reduction and the consolidation of our codebase by letting us avoid duplication, spot the usage of deprecated APIs or internal (non-API) library code, and identify general purpose code in specific projects (such as utility classes) that can be factored out and shared in a core library.
+- With the help of Sourcegraph, we were able to quickly look at all clients of an API and remove unused attributes that lived in different repositories, ultimately simplifying our APIs and speeding up developer iteration time.
+- Our teams are constantly innovating and building new systems, necessitating decomposing and migrating off of older ones. Sourcegraph gives us the ability to search for and refactor references to deprecated services, libraries, URL patterns, and more across our 2000+ repositories, and the confidence that we're not leaving anyone behind.
+
+## Who benefits
+
+<!-- Link to the personas that relate to this use case, and describe how it benefits each of them -->
```
Comment (jasonyavorsky) [inline content/strategy-goals/strategy/use_cases/code_health.md]:
```suggestion
<!-- Link to the personas that relate to this use case, and describe briefly how it benefits each of them (the real detail is in the above section, so be sure not to repeat yourself here; speak in generalities for each persona in this section.) -->
```

**Comment by christinaforney** [comment]
> Anything else you'd add?

Not for this first pass. 

It would probably be helpful if we can help figure out how to structure all the rest of the information (other tools, state without Sourcegraph, etc) so that the entire google doc can end up in the handbook in a way that isn't a massive single handbook page. Perhaps think about how the information will be consumed (the different internal consumers)?

**Comment by jasonyavorsky** [comment]
> > Anything else you'd add?
> 
> Not for this first pass.
> 
> It would probably be helpful if we can help figure out how to structure all the rest of the information (other tools, state without Sourcegraph, etc) so that the entire google doc can end up in the handbook in a way that isn't a massive single handbook page. Perhaps think about how the information will be consumed (the different internal consumers)?

Totally agree @christinaforney - I'll take a todo to work with the use cases working group on that, but I agree all that other info shouldn't slowly filter onto these use case pages, slowly but surely turning them back into 23-pagers.

**Review by nicksnyder — APPROVED** [review]
Love it!

**Pull request merged**